### PR TITLE
Add IAB EU Consent Management Providers (CMP)

### PR DESCRIPTION
**Pull request by andydavies**
Based on TCFv2.0 list from https://iabeurope.eu/cmp-list/

@patrickhulce:

What's the best way to test this, is there anything I should do other than building the project?

Also I've just chosen the IAB EU CMPs right now but wonder if these should really be a sub-category within ads, but also not sure what should be be done with other cookie banners that aren't IAB compliant CMPs

### CODE REVIEW

**Comment by eligrey** [comment]
Cookie consent managers should all be grouped together (IAB or not). IAB compliance isn't a useful signal as there aren't adequate technical limitations for compliance to mean anything important

**Comment by andydavies** [comment]
@eligrey In TCF v2 the IAB CMPs have a commonly defined API which separates them from other cookie consent platforms.

The API allows ad providers to check consent before bidding, so becomes part of the critical path for ads in a way other cookie consents may not. It also gives us a hook to measure their speed in the wild by listening for their loaded signal.

Not saying we shouldn't just group all cookie consents together just that the IAB ones have some useful properties and that people who use this dataset might want that differentiation.

**Comment by eligrey** [comment]
> The API allows ad providers to check consent before bidding, so becomes part of the critical path for ads in a way other cookie consents may not

There are insufficient technical measures to actually enforce that consent is legitimately set via explicit user authorization. This is why I would consider them no better/different (and arguably worse) from many other ad consent sharing platforms.

Ad providers can simply ignore IAB's best practices and 'rules' because their consent sharing platform is held together with legal agreements instead of technical measures. It's not that hard to spoof IAB consent as a malicious ad provider.

File: data/categories.json [hunk 1]
```diff
@@ -60,6 +60,12 @@
     "title": "Tag Management",
     "description": "These scripts tend to load lots of other scripts and initiate many tasks."
   },
+  "cmp": {
```
Comment (patrickhulce) [inline data/categories.json:63]: WDYT about `consent-manager`? CMP pulls double duty in other tech subcircles

File: data/categories.json [hunk 2]
```diff
@@ -60,6 +60,12 @@
...
+    "color": "hsl(345, 90%, 60%)",
+    "title": "Consent Management Provider",
+    "description":
+      "IAB Consent Managment Providers are the 'Cookie Consent' popups used by many publishers. They're invoked for every page and sit on the critical path between a page loading and adverts being displayed."
```
Comment (patrickhulce) [inline data/categories.json:67]:
```suggestion
      "IAB Consent Management Providers are the 'Cookie Consent' popups used by many publishers. They're invoked for every page and sit on the critical path between a page loading and adverts being displayed."
```

File: data/entities.js [hunk 3]
```diff
@@ -11652,4 +11652,400 @@ module.exports = [
...
+    name: 'Next14',
+    homepage: 'https://www.next14.com/',
+    categories: ['cmp'],
+    domains: ['next14.mgr.consensu.org'],
+  },
+  {
+    name: 'AdRoll',
+    homepage: 'https://www.adroll.com/',
+    categories: ['cmp'],
+    domains: ['adroll.mgr.consensu.org'],
```
Comment (patrickhulce) [inline data/entities.js:11881]: is `d.adroll.mgr.consensu.org` not valid? more generally, should we be concerned about subdomains of these being attributed?

**Review by patrickhulce — COMMENTED** [review]
thanks very much @andydavies! 

I think it's fine to leave in the IAB qualifier in the *description* and if/when other non-conforming consent managers are added we can apply subcategories to differentiate at that time. How does that sound?

**Comment by patrickhulce** [comment]
> What's the best way to test this, is there anything I should do other than building the project?

You can add another entry to the jest unit tests that it correctly identifies a particular example resource URL as the consent category, but just building it should be fine too.

Looks like the expectations might need updating, though partially unrelated.

File: data/categories.json (hunk 1 above)
Comment (andydavies) [reply data/categories.json:63]: Yes, think that probably makes more sense - explicit is generally better

File: data/entities.js (hunk 3 above)
Comment (andydavies) [reply data/entities.js:11881]: Just queried the HTTPArchive data and it looks like some CMPs are using deeper subdomains and some aren't, so yes I think we should also add `*.` e.g `*.adroll.mgr.consensu.org` subdomains for attribution too.

**Comment by patrickhulce** [comment]
looks like subdomain wildcard support was never actually included before and that's why the expansion was happening in the first place, hang on one sec :)

**Comment by andydavies** [comment]
Thanks my local install is failing to install v8

There's some duplicates names I still need to address, ATM I'm just adding CMP on the name

**Comment by patrickhulce** [comment]
OK if you rebase I would expect the tests to start passing 👍

**Review by patrickhulce — APPROVED** [review]

**Comment by patrickhulce** [comment]
thanks @andydavies! 🎉

**Comment by HeinzBaumann** [comment]
Thank you so much!

**Pull request merged**